### DISCUSSION TITLE
I do not receive custom push notifications when the application is closed

### DISCUSSION THREAD

**Discussion by experimentosconai** [firebase/flutterfire#11697] [Q&A]
I have tried many times, so finally I want to come to you to find the solution to show the notification when the application is closed, as proof in mainpage.dart there is a Notifications icon, after clicking that icon it is notified in 5 seconds.
- The notification appears in the foreground
- The notification appears in the background
- The notification does not appear when the application is terminated

Within the logic of the icon there are 2 conditions, if the application status is active, the local notification is shown. Otherwise a push notification would be used using the function that comes in fcm_helper.dart.

- I have looked for many solutions to show notifications, so I tried activating autostart or special permissions (autostart in the background), I gave that permission to my application and I had to restart. There are notifications, but that would not be a good practice for the user experience. That's why I want to look for the solution.
- Using Visual Studio Code.
- Using USB physical cellular Xiaomi, Samsung

Attached mainpage.dart, fcm_helper.dart, Manifest, Gradle, main.dart

mainpage.dart:
`import 'package:chatgpt_project/src/fcm/fcm_helper.dart';
//import 'package:chatgpt_project/src/notifications/fcm_schedule_notifications.dart';
import 'package:cloud_firestore/cloud_firestore.dart';
import 'package:flutter/material.dart';

import 'package:firebase_auth/firebase_auth.dart';
import 'package:chatgpt_project/src/authentication/auth_google.dart';
import 'package:chatgpt_project/src/routes/routes.dart';
import 'package:chatgpt_project/src/utils/constants.dart';
import 'package:chatgpt_project/src/views/calendario.dart';
import 'package:chatgpt_project/src/views/inicio.dart';
import 'package:chatgpt_project/src/views/mas.dart';
import 'package:chatgpt_project/src/views/mi_salud.dart';
import 'package:chatgpt_project/src/firestore/user_model.dart';

import 'package:chatgpt_project/src/notifications/schedule_notifications.dart';
import 'package:flutter_local_notifications/flutter_local_notifications.dart';
import 'package:timezone/timezone.dart' as tz;

class MainPage extends StatefulWidget {
  final int initialIndex;
  const MainPage({Key? key, this.initialIndex = 0}) : super(key: key);

  @override
  _MainPageState createState() => _MainPageState();
}

class _MainPageState extends State<MainPage> with WidgetsBindingObserver {
  final FirebaseAuth _auth = FirebaseAuth.instance;
  UserModel? currentUser;
  // 1. Inicialización del plugin de notificaciones locales
  // final FlutterLocalNotificationsPlugin flutterLocalNotificationsPlugin =
  //     FlutterLocalNotificationsPlugin();

  int selectedIndex =
      0; // Indice para controlar la navegación en la barra inferior

  Future<void> _signOut() async {
    await signOutGoogle();
    await _auth.signOut();
    // ignore: use_build_context_synchronously
    Navigator.pushNamedAndRemoveUntil(
      context,
      Routes.login,
      (route) => false,
    );
  }

  void _createNotificationChannel() {
    const AndroidNotificationChannel channel = AndroidNotificationChannel(
      'high_priority_channel', // id
      'High Priority Notifications', // title
      description: 'This channel is used for important notifications.',
      importance: Importance.high,
      playSound: true,
    );

    final FlutterLocalNotificationsPlugin flutterLocalNotificationsPlugin =
        FlutterLocalNotificationsPlugin();

    flutterLocalNotificationsPlugin
        .resolvePlatformSpecificImplementation<
            AndroidFlutterLocalNotificationsPlugin>()
        ?.createNotificationChannel(channel);
  }

  void _sendLocalNotification(tz.TZDateTime scheduledDate) async {
    const AndroidNotificationDetails androidPlatformChannelSpecifics =
        AndroidNotificationDetails(
      'high_priority_channel', // id
      'High Priority Notifications', // title
      importance: Importance.high,
      priority: Priority.high,
      showWhen: false,
      playSound: true,
    );

    const NotificationDetails platformChannelSpecifics =
        NotificationDetails(android: androidPlatformChannelSpecifics);

    final FlutterLocalNotificationsPlugin flutterLocalNotificationsPlugin =
        FlutterLocalNotificationsPlugin();

    await flutterLocalNotificationsPlugin.zonedSchedule(
      0, // ID de notificación
      'Titulo Local',
      'Esta es la descripción de la notificación local',
      scheduledDate,
      platformChannelSpecifics,
      androidAllowWhileIdle: true,
      uiLocalNotificationDateInterpretation:
          UILocalNotificationDateInterpretation.absoluteTime,
    );
  }

  @override
  void initState() {
    super.initState();

    WidgetsBinding.instance.addObserver(this);
    _createNotificationChannel();
    _requestNotificationPermissions();
    _getCurrentUser();
    selectedIndex = widget.initialIndex;
  }

  @override
  void dispose() {
    WidgetsBinding.instance.removeObserver(this);
    super.dispose();
  }

  @override
  void didChangeAppLifecycleState(AppLifecycleState state) {
    super.didChangeAppLifecycleState(state);
    // Aquí puedes manejar los cambios de estado si es necesario
  }

  void _requestNotificationPermissions() {
    FlutterLocalNotificationsPlugin flutterLocalNotificationsPlugin =
        FlutterLocalNotificationsPlugin();

    flutterLocalNotificationsPlugin
        .resolvePlatformSpecificImplementation<
            AndroidFlutterLocalNotificationsPlugin>()
        ?.requestPermission();
  }

  Future<void> _getCurrentUser() async {
    final firebaseUser = _auth.currentUser;
    if (firebaseUser != null) {
      final userDoc = await FirebaseFirestore.instance
          .collection('users')
          .doc(firebaseUser.uid)
          .get();
      setState(() {
        currentUser = UserModel.fromFirestore(userDoc);
      });
      // Llamada a fetchMedicines
      fetchMedicines(firebaseUser.uid);
    }
  }

  @override
  Widget build(BuildContext context) {
    final screens = [
      if (currentUser != null)
        InicioView(user: currentUser!), //aquí pasamos el usuario
      const CalendarioView(),
      if (currentUser != null)
        MiSaludView(
            user:
                currentUser!), //user: currentUser! sirve para conectar con mi_salud.dart
      const MasView()
    ];

    return Scaffold(
      backgroundColor: scaffoldBackgroundColor,
      appBar: AppBar(
        title: const Icon(
          Icons.account_circle,
          color: Colors.white,
          size: 40.0,
        ),
        centerTitle: true,
        backgroundColor: kTextColor,
        elevation: 0,
        automaticallyImplyLeading: true, // Hacer que muestre el botón del menú
        actions: <Widget>[
          Padding(
            padding: const EdgeInsets.only(right: 9.0),
            child: IconButton(
              icon: const Icon(Icons.notifications,
                  color: Colors.white, size: 33.0),
              onPressed: () async {
                AppLifecycleState? currentState =
                    WidgetsBinding.instance.lifecycleState;
                try {
                  if (currentState == AppLifecycleState.resumed ||
                      currentState == AppLifecycleState.inactive) {
                    // La aplicación está en primer plano o en segundo plano. Envía notificación local.
                    tz.TZDateTime scheduledDate = tz.TZDateTime.now(tz.local)
                        .add(const Duration(seconds: 5));
                    _sendLocalNotification(scheduledDate);
                    print('Notificación local programada.');
                  } else if (currentState == AppLifecycleState.detached) {
                    // La aplicación está completamente cerrada. Envía notificación push.
                    String? currentUserId =
                        FirebaseAuth.instance.currentUser?.uid;
                    if (currentUserId != null) {
                      DocumentSnapshot userDoc = await FirebaseFirestore
                          .instance
                          .collection('users')
                          .doc(currentUserId)
                          .get();
                      String? token =
                          (userDoc.data() as Map<String, dynamic>)['fcmToken'];
                      if (token != null) {
                        print('Preparando para enviar notificación push...');
                        await sendPushNotification(
                            token,
                            'Notificación de prueba',
                            'Esta es una notificación de prueba');
                        print('Notificación push enviada exitosamente.');
                      } else {
                        print('No se pudo obtener el token FCM.');
                      }
                    } else {
                      print('No hay usuario autenticado.');
                    }
                  }
                } catch (e) {
                  print("Error al intentar enviar notificación: $e");
                }
              },
            ),
          ),
        ],
      ),
      drawer: Drawer(
        child: ListView(
          padding: EdgeInsets.zero,
          children: <Widget>[
            const DrawerHeader(
              decoration: BoxDecoration(
                color: kTextColor,
              ),
              child: Text(
                'Menu',
                style: TextStyle(
                  color: Colors.white,
                  fontSize: 24,
                ),
              ),
            ),
            ListTile(
              leading: const Icon(Icons.logout),
              title: const Text('Cerrar sesión'),
              onTap: _signOut,
            ),
          ],
        ),
      ),
      body: IndexedStack(
        index: selectedIndex,
        children: screens,
      ),
      bottomNavigationBar: BottomNavigationBar(
        currentIndex: selectedIndex,
        onTap: (index) {
          // Solo actualiza el estado si el índice es diferente
          setState(() {
            selectedIndex = index;
          });
        },
        items: const [
          BottomNavigationBarItem(
            icon: Icon(Icons.home_outlined),
            activeIcon: Icon(Icons.home), //cambia de icono al seleccionar
            label: 'Inicio',
            backgroundColor: kTextColor,
          ),
          BottomNavigationBarItem(
            icon: Icon(Icons.calendar_month_outlined),
            activeIcon: Icon(Icons.calendar_month),
            label: 'Calendario',
            backgroundColor: kTextColor,
          ),
          BottomNavigationBarItem(
            icon: Icon(Icons.monitor_heart),
            activeIcon: Icon(Icons.monitor_heart),
            label: 'Mi Salud',
            backgroundColor: kTextColor,
          ),
          BottomNavigationBarItem(
            icon: Icon(Icons.more_horiz_outlined),
            activeIcon: Icon(Icons.more_horiz),
            label: 'Más',
            backgroundColor: kTextColor,
          ),
        ],
      ),
    );
  }
}
`

fcm_helper.dart:
`Future<String> getAccessToken() async {
  try {
    final client = http.Client();
    final AccessCredentials credentials =
        await obtainAccessCredentialsViaServiceAccount(_credentials,
            ['https://www.googleapis.com/auth/firebase.messaging'], client);

    final accessToken = credentials.accessToken.data;
    client.close();
    return accessToken;
  } catch (e) {
    print("Error obteniendo el token de acceso: $e");
    throw e;
  }
}

Future<void> sendPushNotification(
    String token, String title, String body) async {
  try {
    const String sendUrl =
        'https://fcm.googleapis.com/v1/projects/proyecto-chatgpt-84b99/messages:send';
    final String accessToken = await getAccessToken();

    final response = await http.post(
      Uri.parse(sendUrl),
      headers: {
        'Authorization': 'Bearer $accessToken',
        'Content-Type': 'application/json',
      },
      body: json.encode({
        'message': {
          'token': token,
          'notification': {
            'title': title,
            'body': body,
          },
          'android': {
            'priority': 'high',
          },
        },
      }),
    );

    if (response.statusCode != 200) {
      throw Exception('Failed to send notification: ${response.body}');
    }
  } catch (e) {
    print("Error sending the notification: $e");
    throw e;
  }
}`

AndroidManifest.xml:
`<manifest xmlns:android="http://schemas.android.com/apk/res/android"
    package="com.example.chatgpt_project">

   
   <uses-permission android:name="android.permission.INTERNET" />

   
   <uses-permission android:name="android.permission.RECEIVE_BOOT_COMPLETED"/>
   <uses-permission android:name="android.permission.VIBRATE" />
   <uses-permission android:name="android.permission.POST_NOTIFICATIONS" />
   <uses-permission android:name="android.permission.WAKE_LOCK" />
   <uses-permission android:name="android.permission.FOREGROUND_SERVICE" />

   <application
        android:label="chatgpt_project"
        android:name="${applicationName}"
        android:icon="@mipmap/ic_launcher">
        <activity
            android:name=".MainActivity"
            android:exported="true"
            android:launchMode="singleTop"
            android:theme="@style/LaunchTheme"
            android:configChanges="orientation|keyboardHidden|keyboard|screenSize|smallestScreenSize|locale|layoutDirection|fontScale|screenLayout|density|uiMode"
            android:hardwareAccelerated="true"
            android:windowSoftInputMode="adjustResize">
            
            <meta-data
              android:name="io.flutter.embedding.android.NormalTheme"
              android:resource="@style/NormalTheme"
              />
            <intent-filter>
                <action android:name="android.intent.action.MAIN"/>
                <category android:name="android.intent.category.LAUNCHER"/>
            </intent-filter>
        </activity>
        
        <meta-data
            android:name="flutterEmbedding"
            android:value="2" />
        
        <receiver android:name="com.dexterous.flutterlocalnotifications.ScheduledNotificationBootReceiver" android:exported="false">
            <intent-filter>
                <action android:name="android.intent.action.BOOT_COMPLETED"></action>
                <action android:name="android.intent.action.MY_PACKAGE_REPLACED"></action>
            </intent-filter>
        </receiver>
        <receiver android:name="com.dexterous.flutterlocalnotifications.ScheduledNotificationReceiver" android:exported="false" />

        <receiver android:name=".BootCompletedReceiver" android:exported="false">
            <intent-filter>
                <action android:name="android.intent.action.BOOT_COMPLETED" />
            </intent-filter>
        </receiver>
        
        <receiver android:name=".BootReceiver" android:enabled="true" android:exported="true">
            <intent-filter>
                <action android:name="android.intent.action.BOOT_COMPLETED"/>
                <category android:name="android.intent.category.DEFAULT" />
            </intent-filter>
        </receiver>

        
        <meta-data
        android:name="com.google.firebase.messaging.default_notification_channel_id"
        android:value="high_priority_channel" />
    </application>
</manifest>
`

build.gradle:
`def localProperties = new Properties()
def localPropertiesFile = rootProject.file('local.properties')
if (localPropertiesFile.exists()) {
    localPropertiesFile.withReader('UTF-8') { reader ->
        localProperties.load(reader)
    }
}

def flutterRoot = localProperties.getProperty('flutter.sdk')
if (flutterRoot == null) {
    throw GradleException("Flutter SDK not found. Define location with flutter.sdk in the local.properties file.")
}

def flutterVersionCode = localProperties.getProperty('flutter.versionCode')
if (flutterVersionCode == null) {
    flutterVersionCode = '1'
}

def flutterVersionName = localProperties.getProperty('flutter.versionName')
if (flutterVersionName == null) {
    flutterVersionName = '1.0'
}

apply plugin: 'com.android.application'
apply plugin: 'kotlin-android'
apply from: "$flutterRoot/packages/flutter_tools/gradle/flutter.gradle"

android {
    compileSdkVersion 33
    //compileSdkVersion flutter.compileSdkVersion
    ndkVersion flutter.ndkVersion

    compileOptions {
        // Flag to enable support for the new language APIs
        coreLibraryDesugaringEnabled true
        // Sets Java compatibility to Java 8
        sourceCompatibility JavaVersion.VERSION_1_8
        targetCompatibility JavaVersion.VERSION_1_8
    }

    kotlinOptions {
        jvmTarget = '1.8'
    }

    sourceSets {
        main.java.srcDirs += 'src/main/kotlin'
    }

    defaultConfig {
        // TODO: Specify your own unique Application ID (https://developer.android.com/studio/build/application-id.html).
        applicationId "com.example.chatgpt_project"
        // You can update the following values to match your application needs.
        // For more information, see: https://docs.flutter.dev/deployment/android#reviewing-the-gradle-build-configuration.
        minSdkVersion 21
        targetSdkVersion 33
        versionCode flutterVersionCode.toInteger()
        versionName flutterVersionName
        multiDexEnabled true
    }

    buildTypes {
        release {
            // TODO: Add your own signing config for the release build.
            // Signing with the debug keys for now, so `flutter run --release` works.
            signingConfig signingConfigs.debug
        }
    }

    lintOptions {
        disable 'InvalidPackage'
        disable "Instantiatable"
        checkReleaseBuilds false
        abortOnError false
    }
}

flutter {
    source '../..'
}

dependencies {
    implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
    implementation 'com.android.support:multidex:2.0.1'
    coreLibraryDesugaring 'com.android.tools:desugar_jdk_libs:1.1.5'
    implementation "androidx.window:window:1.0.0"
    implementation "androidx.window:window-java:1.0.0"
}

main.dart:
`import 'dart:ui';`

import 'package:chatgpt_project/src/authentication/auth_google.dart';
//import 'package:chatgpt_project/src/fcm/fcm_helper.dart';
import 'package:chatgpt_project/src/notifications/schedule_notifications.dart';
import 'package:cloud_firestore/cloud_firestore.dart';
import 'package:flutter/material.dart';

import 'package:firebase_messaging/firebase_messaging.dart';

import 'package:chatgpt_project/src/routes/routes.dart';
import 'package:firebase_core/firebase_core.dart';
import 'package:flutter_local_notifications/flutter_local_notifications.dart';
import 'firebase_options.dart';
//import 'package:firebase_analytics/firebase_analytics.dart';
import 'package:intl/date_symbol_data_local.dart';
import 'package:provider/provider.dart';
import 'package:chatgpt_project/src/providers/chats_provider.dart';
import 'package:firebase_auth/firebase_auth.dart';

//import 'package:chatgpt_project/src/providers/models_provider.dart';
//import 'package:flutter_dotenv/flutter_dotenv.dart';
import 'package:timezone/data/latest.dart' as tz;
import 'package:timezone/timezone.dart' as tz;
import 'package:flutter/services.dart';
import 'package:shared_preferences/shared_preferences.dart';
import 'package:flutter_native_timezone/flutter_native_timezone.dart';

final FirebaseFirestore _firestore = FirebaseFirestore.instance;

Future<void> firebaseMessagingBackgroundHandler(RemoteMessage message) async {
  print('Mensaje recibido en segundo plano: ${message.messageId}');

  const AndroidNotificationDetails androidPlatformChannelSpecifics =
      AndroidNotificationDetails(
    'high_priority_channel', // id
    'High Priority Notifications', // title
    importance: Importance.max,
    priority: Priority.high,
    showWhen: false,
  );

  const NotificationDetails platformChannelSpecifics =
      NotificationDetails(android: androidPlatformChannelSpecifics);

  await FlutterLocalNotificationsPlugin().show(
    0, // ID de notificación
    'Titulo Prueba',
    'Esta es la descripcion Prueba',
    platformChannelSpecifics,
    payload: 'Default_Sound',
  );
}

void main() async {
  WidgetsFlutterBinding.ensureInitialized();
  tz.initializeTimeZones();

  final String timeZoneName = await FlutterNativeTimezone.getLocalTimezone();
  tz.setLocalLocation(tz.getLocation(timeZoneName));

  FirebaseMessaging.onBackgroundMessage(firebaseMessagingBackgroundHandler);

  await Firebase.initializeApp(
    options: DefaultFirebaseOptions.currentPlatform,
  );

  await initializeDateFormatting('es_ES', null);

  _configureFirebaseListeners();

  final CallbackHandle callback =
      PluginUtilities.getCallbackHandle(callbackDispatcher)!;
  final int callbackHandle = callback.toRawHandle();
  final SharedPreferences prefs = await SharedPreferences.getInstance();
  await prefs.setInt('callback_handle', callbackHandle);

  runApp(
    MultiProvider(
      providers: [
        ChangeNotifierProvider(create: (context) => ChatProvider()),
      ],
      child: const MyApp(),
    ),
  );
}

Future<void> updateTokenToFirestore(String token) async {
  // Obtiene el userId del usuario autenticado
  String? currentUserId = FirebaseAuth.instance.currentUser?.uid;

  if (currentUserId != null) {
    final docRef = _firestore.collection('users').doc(currentUserId);
    final docSnapshot = await docRef.get();

    if (docSnapshot.exists) {
      await docRef.update({'fcmToken': token});
    } else {
      // Considera la posibilidad de crear un nuevo documento si no existe.
      // Puedes agregar más campos si es necesario.
      await docRef.set({
        'fcmToken': token,
        // 'otrosCampos': 'otrosValores',
      });
    }
  }
}

void _configureFirebaseListeners() async {
  final FirebaseMessaging _firebaseMessaging = FirebaseMessaging.instance;

  // Obtener y manejar el token de FCM
  _firebaseMessaging.getToken().then((String? token) {
    assert(token != null);
    print("Token de FCM: $token");
    updateTokenToFirestore(token!);
  });

  // Escuchar cambios en el token
  _firebaseMessaging.onTokenRefresh.listen((String token) {
    print("Token de FCM actualizado: $token");
    updateTokenToFirestore(token);
  });

  _firebaseMessaging.getInitialMessage().then((RemoteMessage? message) {
    if (message != null) {
    }
  });

  FirebaseMessaging.onMessage.listen((RemoteMessage message) {
  });

  FirebaseMessaging.onMessageOpenedApp.listen((RemoteMessage message) {
  });
}

void fetchMedicinesFromBoot(List<String> args) async {
  // Aquí puedes invocar fetchMedicines con el userId que necesites.
  String? userId =
      await getUserId(); // Aquí deberías implementar la función getUserId si aún no lo has hecho.
  if (userId != null) {
    fetchMedicines(userId);
  }
}

void callbackDispatcher() {
  const MethodChannel _backgroundChannel =
      MethodChannel('com.example.chatgpt_project/background_channel');
  WidgetsFlutterBinding.ensureInitialized();

  _backgroundChannel.setMethodCallHandler((MethodCall call) async {
    if (call.method == 'fetchMedicinesFromBoot') {
      fetchMedicinesFromBoot([]);
    }
  });
}

class MyApp extends StatelessWidget {
  const MyApp({Key? key})
      : super(
            key:
                key); //CON ANALYTICS: const MyApp({Key? key, required this.analytics}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'My App',
      debugShowCheckedModeBanner: false,
      initialRoute: Routes.loading,
      routes: Routes.routes,
    );
  }
}`

flutter doctor -v
`[√] Flutter (Channel stable, 3.13.6, on Microsoft Windows [Versi¢n 10.0.19045.3448], locale es-PE)
    • Flutter version 3.13.6 on channel stable at C:\src\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision ead455963c (2 weeks ago), 2023-09-26 18:28:17 -0700
    • Engine revision a794cf2681
    • Dart version 3.1.3
    • DevTools version 2.25.0

[√] Windows Version (Installed version of Windows is version 10 or higher)

[√] Android toolchain - develop for Android devices (Android SDK version 33.0.2)
    • Android SDK at C:\Users\User\AppData\Local\Android\sdk
    • Platform android-33-ext5, build-tools 33.0.2
    • Java binary at: C:\Program Files\Android\Android Studio\jbr\bin\java
    • Java version OpenJDK Runtime Environment (build 11.0.15+0-b2043.56-9505619)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop Windows apps (Visual Studio Build Tools 2022 17.5.3)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools
    • Visual Studio Build Tools 2022 version 17.5.33516.290
    • Windows 10 SDK version 10.0.22621.0

[√] Android Studio (version 2022.1)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.15+0-b2043.56-9505619)

[√] VS Code (version 1.83.0)
    • VS Code at C:\Users\User\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.74.0

[√] Connected device (4 available)
    • sdk gphone64 x86 64 (mobile) • emulator-5554 • android-x64    • Android 13 (API 33) (emulator)
    • Windows (desktop)            • windows       • windows-x64    • Microsoft Windows [Versi¢n 10.0.19045.3448]
    • Chrome (web)                 • chrome        • web-javascript • Google Chrome 117.0.5938.150
    • Edge (web)                   • edge          • web-javascript • Microsoft Edge 117.0.2045.60

[√] Network resources
    • All expected network resources are available.

• No issues found!`

` `

**Comment by experimentosconai** [comment]
AndroidManifest.xml:
<manifest xmlns:android="http://schemas.android.com/apk/res/android"
    package="com.example.chatgpt_project">




   <application
        android:label="chatgpt_project"
        android:name="${applicationName}"
        android:icon="@mipmap/ic_launcher">
        <activity
            android:name=".MainActivity"
            android:exported="true"
            android:launchMode="singleTop"
            android:theme="@style/LaunchTheme"
        android:configChanges="orientation|keyboardHidden|keyboard|screenSize|smallestScreenSize|locale|layoutDirection|fontScale|screenLayout|density|uiMode"
            android:hardwareAccelerated="true"
            android:windowSoftInputMode="adjustResize">
          
            <intent-filter>
                <action android:name="android.intent.action.MAIN"/>
                <category android:name="android.intent.category.LAUNCHER"/>
            </intent-filter>
        </activity>
    
        <meta-data
            android:name="flutterEmbedding"
            android:value="2" />
   
        <receiver android:name="com.dexterous.flutterlocalnotifications.ScheduledNotificationBootReceiver" android:exported="false">

            <intent-filter>
                <action android:name="android.intent.action.BOOT_COMPLETED"></action>
                <action android:name="android.intent.action.MY_PACKAGE_REPLACED"></action>
            </intent-filter>

        </receiver>
        <receiver android:name="com.dexterous.flutterlocalnotifications.ScheduledNotificationReceiver" android:exported="false" />
        <receiver android:name=".BootCompletedReceiver" android:exported="false">
            <intent-filter>
                <action android:name="android.intent.action.BOOT_COMPLETED" />
            </intent-filter>
        </receiver>
        <receiver android:name=".BootReceiver" android:enabled="true" android:exported="true">
            <intent-filter>
                <action android:name="android.intent.action.BOOT_COMPLETED"/>
                <category android:name="android.intent.category.DEFAULT" />
            </intent-filter>
        </receiver>
        <meta-data
        android:name="com.google.firebase.messaging.default_notification_channel_id"
        android:value="high_priority_channel" />
    </application>
</manifest>


   <uses-permission android:name="android.permission.INTERNET" />

   <uses-permission android:name="android.permission.RECEIVE_BOOT_COMPLETED"/>
   <uses-permission android:name="android.permission.VIBRATE" />
   <uses-permission android:name="android.permission.POST_NOTIFICATIONS" />
   <uses-permission android:name="android.permission.WAKE_LOCK" />
   <uses-permission android:name="android.permission.FOREGROUND_SERVICE" />

**Comment by pelumibabs** [comment]
Did you tried it with debug, or release mode? I experienced the same when I started my app in Debug mode.

**Reply by experimentosconai** [reply]
I am using flutter run --release, notifications do not work until I press the Autostart or sbackground permission to the application and I must restart to apply those changes, can you review my missing code?
I have used flutter notification local and push notification,
